### PR TITLE
zee/steps: depends_on gmsh+mpi

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: deploy/packages/external-libraries.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ packages:
       - cudnn@7.3.0
       # - freeimage@3.18.0
       - glew
-      - gmsh+metis~mpi+oce+openmp+shared@4.4.1
+      - gmsh+metis+mpi+oce+openmp+shared@4.4.1
```
Comment (matz-e) [inline deploy/packages/external-libraries.yaml]: Once we depend on `mpi`, could we move this into `parallel-libraries`?

In `external` stuff, the dependency tree gets truncated, and no information about which MPI was used will be propagated…

**Comment by tristan0x** [comment]
Is there something I can do to fix the CI @matz-e ?

**Comment by tristan0x** [comment]
> Is there something I can do to fix the CI @matz-e ?

I don't get the error message in failing CI: "Error: Package gmsh does not depend on hpe-mpi"
There is no occurrence of gmsh without MPI in `deploy/` directory.
Can you please take a look at it @matz-e and @pramodk ?

**Comment by pramodk** [comment]
> I don't get the error message in failing CI: "Error: Package gmsh does not depend on hpe-mpi"

@tristan0x : my best guess is : you initially put `gmsh+metis+mpi` in external libraries and then moved to parallel libraries. All packages in external libraries "loose" their dependency tree as they are configured as "external packages". Hence, when gmsh is used downstream in later stages (e.g. application), gmsh says I am not dependent on mpi!

I am going to remove PR directory and re-trigger CI. Lets see if that improves the situation!

**Comment by pramodk** [comment]
Retest this please

**Comment by pramodk** [comment]
I am looking at the BB5 CI and I see:

```

[2020-12-10T23:32:41.983Z] ==> Installing oce
[2020-12-10T23:32:41.983Z] ==> Searching for binary cache of oce
[2020-12-10T23:32:41.983Z] ==> Finding buildcaches in /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/980/mirror/binaries/build_cache
[2020-12-10T23:32:41.983Z] ==> Finding buildcaches in /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/980/mirror/sources/build_cache
[2020-12-10T23:32:41.983Z] ==> No binary for oce found: installing from source
[2020-12-10T23:33:00.387Z] ==> Fetching file:///gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/980/mirror/sources/oce/oce-0.18.3.tar.gz
[2020-12-10T23:33:00.388Z] ==> Staging archive: /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/980/tmp/bbprelman/spack-stage-oce-0.18.3-hwirrrnciiniamic5vm74or4f4qbl5t2/OCE-0.18.3.tar.gz
[2020-12-10T23:33:00.388Z] ==> Created stage in /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/980/tmp/bbprelman/spack-stage-oce-0.18.3-hwirrrnciiniamic5vm74or4f4qbl5t2
[2020-12-10T23:33:00.388Z] ==> No patches needed for oce
[2020-12-10T23:33:00.388Z] ==> Building oce [Package]
[2020-12-10T23:33:00.388Z] ==> Executing phase: 'install'
[2020-12-11T06:01:19.464Z] ==> Successfully installed oce
[2020-12-11T06:01:19.464Z]   Fetch: 0.04s.  Build: 6h 28m 24.08s.  Total: 6h 28m 24.12s.
[2020-12-11T06:01:19.464Z] [+] /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/980/deploy/libraries/2020-12-11/linux-rhel7-x86_64/gcc-8.3.0/oce-0.18.3-hwirrrncii
```

i.e. OCE package took 6h 30min to build! Is this normal @tristan0x ? and if so, is that dependency required?

File: deploy/packages/external-libraries.yaml [hunk 2]
```diff
@@ -28,7 +28,6 @@ packages:
       - cudnn@7.3.0
       # - freeimage@3.18.0
       - glew
-      - gmsh+metis~mpi+oce+openmp+shared@4.4.1
```
Comment (matz-e) [inline deploy/packages/external-libraries.yaml:31]: This has moved to `environments/externals.yaml`

File: deploy/packages/parallel-libraries.yaml [hunk 3]
```diff
@@ -46,6 +46,7 @@ packages:
       - mpi
       - lapack
     specs:
+      - gmsh+metis+mpi+oce+openmp+shared@4.4.1
```
Comment (matz-e) [inline deploy/packages/parallel-libraries.yaml:49]: As above, now in `environments/libraries.yaml`

File: deploy/packages/external-libraries.yaml (hunk 2 above)
Comment (tristan0x) [reply deploy/packages/external-libraries.yaml:31]: I updated `environment/externals.yaml`. Should I revert the initial changes of this PR @matz-e ?

**Comment by matz-e** [comment]
Outdated.

**Pull request closed without merging**